browser/ui/brave_wallet/BUILD.gn

```diff
@@ -0,0 +1,7 @@
+import("//brave/components/brave_wallet/common/buildflags/buildflags.gni")
+
+assert(brave_wallet_enabled && !is_android && !is_ios)
```

Comment:
you have `!is_android && !is_ios` as assertions here, but not where you are including the code like https://github.com/brave/brave-core/pull/9443/commits/edac70b12fb0c79a6654208ec57713265386b766#diff-d665c90359b8b4eb62a748f6fab0151aa41ed2b3cd3de704fd320ba176cd51e6R67
the assertions in gn and the buildflags must match. This means that the tab_helper must either also be subject to the same conditions or else the references to this code from the tab helper must have OS guards

Reply:
Add OS guards in tab helper,
https://github.com/brave/brave-core/pull/9443/commits/39529121ec8feba7ceb823c6ba12861e24d5c6ff